jetstream/analysis.py

```diff
@@ -457,3 +446,42 @@ def run(self, current_date: datetime, dry_run: bool = False) -> None:
 
         result_futures = client.compute(results)
         client.gather(result_futures)  # block until futures have finished
+
+    def ensure_enrollments(
+        self, current_date: datetime, recreate_enrollments: bool = False
+    ) -> None:
+        """Ensure that enrollment tables for experiment are up-to-date or re-create."""
+        time_limits = self._get_timelimits_if_ready(AnalysisPeriod.DAY, current_date)
+
+        if time_limits is None:
+            logger.info("Skipping %s (%s); not ready", self.config.experiment.normandy_slug)
+            return
+
+        if self.config.experiment.start_date is None:
+            raise errors.NoStartDateException(self.config.experiment.normandy_slug)
+
+        normalized_slug = bq_normalize_name(self.config.experiment.normandy_slug)
+        enrollments_table = f"enrollments_{normalized_slug}"
+
+        if not recreate_enrollments:
+            # check if enrollments table already exists and skip creation
+            try:
+                self.bigquery.client.get_table(enrollments_table)
```

Comment:
Instead of asking permission we could ask forgiveness; instead of testing whether the table exists, we could just try to run the query using the WRITE_EMPTY disposition and ignore any `google.cloud.exceptions.Conflict` exception that arises

Reply:
also I think it's probably healthy to treat `BigQueryClient.client` as a private member when we can, to provide a little bit of API isolation